### ISSUE TITLE
chinese unicode 

### ISSUE THREAD

**Issue by whaqzhzd** [mapeditor/tiled-extensions#9]
After I encode Chinese in unicode in my script, I save the single slash, but the file is double slash, which causes an error in the encoding of the Chinese content, how should I save the Chinese content correctly?

![image](https://user-images.githubusercontent.com/16913061/135788517-32a7e62f-3f32-4a77-bf43-2418708499d7.png)
![image](https://user-images.githubusercontent.com/16913061/135788533-81c13b72-bf7c-4a47-ada2-3fe1c2a6fbc6.png)
![image](https://user-images.githubusercontent.com/16913061/135788545-f40766a8-9c8e-4982-b4e8-3533ace2afc7.png)

**Comment by bjorn** [comment]
Isn't the double slash simply the result of the JSON encoding? If you load this JSON file and print out the string, it should again have only a single slash.

**Comment by whaqzhzd** [comment]
Yes, if you don't encode it this way, the output will be garbled after the file

![image](https://user-images.githubusercontent.com/16913061/135984186-f63eec8e-0400-42c8-8431-f23af3e52018.png)

**Comment by whaqzhzd** [comment]
Re-introduce the json file with the double slash, neither the tild editor nor the js can handle this file correctly

![image](https://user-images.githubusercontent.com/16913061/135985739-137eec2f-9def-4c99-8a5b-48979fec36b0.png)

![image](https://user-images.githubusercontent.com/16913061/135985785-48cd210a-bd36-41b6-b33d-0d5df3ce83cc.png)

![image](https://user-images.githubusercontent.com/16913061/135985826-bd4e4949-7301-4280-83d0-078b2d32caa1.png)

But the single slash can be correctly parsed

![image](https://user-images.githubusercontent.com/16913061/135985987-717f135a-c2ee-4089-b581-e9bbf3875e2a.png)

![image](https://user-images.githubusercontent.com/16913061/135986030-150cb7ad-1235-49fb-bf69-e522f055bc17.png)

![image](https://user-images.githubusercontent.com/16913061/135986241-e671f32c-d914-40a7-9973-074c9ffed79c.png)

**Comment by bjorn** [comment]
> Yes, if you don't encode it this way, the output will be garbled after the file

I guess there is some encoding bug we need to fix. Could you provide the full script, or at least a small functional one that reproduces the issue?

**Comment by whaqzhzd** [comment]
Sure, with pleasure!

**Comment by whaqzhzd** [comment]
@bjorn This file  [format.zip](https://github.com/mapeditor/tiled-extensions/files/7284282/format.zip)

**Comment by bjorn** [comment]
@whaqzhzd Hmm, regardless of the unicode issue, why are you writing a JS script that exports to basically the same format as the JSON plugin? In the script I read:
```js
//  The main purpose of this script is to restore the
//  exported json format using a js script. So that 
//  developers can easily and quickly develop the map 
//  twice
```
However, if backwards compatibility is a problem with the changes made to the JSON format since Tiled 1.2, you can switch to the "json1" plugin in the Preferences to use the format that was used until Tiled 1.1. Would that be a solution for you?

**Comment by whaqzhzd** [comment]
we will exclude object level data except that the export format is the same as json. Then reassemble the data to export to a new file, such as pathfinding data, and we will build the graph data in the script to better control the map made by tiled in the development game. Of course there are other features. Such as streamlining the format or generating configuration files more easily.

**Comment by whaqzhzd** [comment]
@bjorn ^_^
